figures/scripts/fig5-panels-gsva-umap.R

```diff
@@ -260,28 +261,49 @@ gsva_heatmap <- Heatmap(
   na_col = na_color$hex_codes,
   show_column_names = FALSE,
   cluster_columns = FALSE,
-  row_names_gp = grid::gpar(fontsize = 5),
+  row_names_gp = grid::gpar(fontsize = 5.8),
   top_annotation = column_heatmap_annotation,
-  heatmap_legend_param = list(direction = "horizontal")
+  heatmap_legend_param = list(direction = "vertical")
 )
 
-pdf(gsva_pdf, width = 5.75, height = 5)
-draw(gsva_heatmap, heatmap_legend_side = "bottom")
+pdf(gsva_pdf, width = 7, height = 4)
+draw(gsva_heatmap, heatmap_legend_side = "right")
 dev.off()
 
 
 
 #### broad_histology legend ----------------------------------------------------
 
 # UMAP legend
-display_group_legend <- Legend(
-  labels = names(annotation_colors_bh),
-  legend_gp = gpar(fill = annotation_colors_bh)
+# note that legend text cannot be resized in `Legend()`, only the title text
+# Legend() is not great at text wrapping so we're using a hack
+
+histology_order <- names(annotation_colors_bh)
+
+display_group_legend_plot <- tibble::as_tibble(annotation_colors_bh, 
+                  rownames = "histology") %>%
+  mutate(histology = factor(histology, 
+                            levels = names(annotation_colors_bh))) %>%
+  ggplot() + 
+  aes(x = annotation_colors_bh, y = annotation_colors_bh, color = histology) + 
+  geom_point(size = 3.75) + 
+  scale_color_manual(values = annotation_colors_bh, 
+                     labels = function(x) str_wrap(x, 20)) + 
+  ggpubr::theme_pubr() +
+  guides(color = guide_legend(ncol = 3)) +
+  theme(
+    legend.title = element_blank(),
+    legend.text = element_text(size = 8.5), 
+)
+  
+display_group_legend <- cowplot::get_legend(display_group_legend_plot)
+
+cowplot::save_plot(
+  umap_legend_pdf, 
+  cowplot::ggdraw(display_group_legend), 
+  base_width = 4.75, base_height = 1.25
```

Comment:
You are assembling this legend separately and joining illustrator, right? Can you squeeze them together a bit more? (Probably by making the UMAP a bit larger)

Reply:
Hm, maybe I should join with code... on it either way!

Reply:
Noting that given https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1753, I'm going to keep these as separate exports so the legend can still be popped into S7.